x/mongo/driver/topology/topology.go

```diff
@@ -312,6 +313,11 @@ func (t *Topology) SelectServer(ctx context.Context, ss description.ServerSelect
 		var suitable []description.Server
 		var selectErr error
 
+		desc := t.Description()
```

Comment:
Below, we do server selection by either calling `selectServerFromDescription` or `selectServerFromSubscription`. The `selectServerFromSubscription` method also loops and internally calls `selectServerFromDescription` every time the topology description is updated, so doing the compatibility error check up here isn't quite sufficient. I think we can move this check into `selectServerFromDescription`. It's already got a topology description and doesn't block at all, so it seems like the ideal place.

Reply:
done